### PR TITLE
PP-450/optimise-gas-consumption

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Removing feeReceiver
- Changes on methods to optimise gas consumption 

## Why

- The primary objective of RIF Relay is to relay transaction the cheapest way possible.

## Refs

- [Jira](https://rsklabs.atlassian.net/browse/PP-450)

### CODE REVIEW

File: contracts/Collector.sol [hunk 1]
```diff
@@ -72,14 +69,14 @@ contract Collector is ICollector{
...
     noBalanceToShare
     {
-        address oldRemainderAddress = remainderAddress;
-        remainderAddress = _remainderAddress;
-
         uint256 balance = token.balanceOf(address(this));
 
         if(balance != 0) {
-            token.transfer(oldRemainderAddress, balance);
+            token.transfer(remainderAddress, balance);
```
Comment (jurajpiar) [inline contracts/Collector.sol:75]:
```suggestion
            token.transfer(_remainderAddress, balance);
```
I'm guessing?

**Review by ironFe93 — APPROVED** [review]
lgtm

**Comment by franciscotobar** [comment]
We will be closing this PR since by removing the feesReceiver we are introducing a new risk.

**Comment by antomor** [comment]
@franciscotobar Should we close also the related PRs referenced above?

**Comment by franciscotobar** [comment]
The PR's were closed :)

**Pull request closed without merging**